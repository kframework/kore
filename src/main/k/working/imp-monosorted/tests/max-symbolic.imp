// This program calculates in `max`
// the maximum of `x`, `y`, and `z`.

int x, y, z, max;
x = symInt;
y = symInt;
z = symInt;
max = AId x;
if (AId y <= AId max) {max = AId y;} else {}
if (AId y <= AId max) {max = AId z;} else {}
