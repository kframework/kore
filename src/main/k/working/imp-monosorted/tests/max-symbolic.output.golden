    true
  #Equals
    varvarVar'Ques'I1832 <=Int varvarVar'Ques'I2462
#And
    true
  #Equals
    varvarVar'Ques'I2462 <=Int varvarVar'Ques'I3095
#And
  <generatedTop>
    <T>
      <k>
        .
      </k>
      <state>
        x |-> varvarVar'Ques'I1832 y |-> varvarVar'Ques'I2462 z |-> varvarVar'Ques'I3095 max |-> varvarVar'Ques'I3095 .State
      </state>
    </T>
    <generatedCounter>
      0
    </generatedCounter>
  </generatedTop>
